### PR TITLE
Generate concatenation overloads to filter out empty captures.

### PR DESCRIPTION
**Pull request by rxwei**
```swift
let regex = Regex {
  "a".+
  OneOrMore(Character("b")).capture()
  Repeat("c").capture()
  CharacterClass.hexDigit.capture().*
  "e".?
  ("t" | "k").capture()
}
// Regex<([Character], [Substring], [Character], Substring)>
```

### CODE REVIEW

File: Sources/VariadicsGenerator/VariadicsGenerator.swift [hunk 1]
```diff
@@ -0,0 +1,186 @@
...
+  func emitDeclarations(arity: Int) {
+    for permutation in Permutations(arity: arity) {
+      // Emit type declarations.
+      let typeName = "\(concatenationStructTypeBaseName)\(arity)_\(permutation.identifier)"
+      output("public struct \(typeName)<")
+      for i in 0..<arity {
+        output("T\(i): \(patternProtocolName)")
+        if i != arity - 1 {
+          output(", ")
+        }
```
Comment (milseman) [inline Sources/VariadicsGenerator/VariadicsGenerator.swift:100]: I would definitely extract this out into a helper method. I know this is a throw-away script, but it should make the function easier to read and I (side project) want to keep improving string formatting.

Comment (rxwei) [reply Sources/VariadicsGenerator/VariadicsGenerator.swift]: Done!

**Comment by rxwei** [comment]
@swift-ci please test

**Comment by rxwei** [comment]
Looks like the linux bot got stuck.

**Comment by rxwei** [comment]
All tests pass for me locally on macOS.

**Pull request merged**